design/0003-byop.md

```diff
@@ -0,0 +1,198 @@
+# Bring your own Prometheus
+
+- @pothulapati
+- Start Date: 2020-04-15
+- Target Date: 
+- RFC PR: [linkerd/rfc#16](https://github.com/linkerd/rfc/pull/16)
+- Linkerd Issue: [linkerd/linkerd2#3590](https://github.com/linkerd/linkerd2/issues/3590)
+- Reviewers: @grampelberg @alpeb @siggy
+
+## Summary
+
+[summary]: #summary
+
+This RFC aims to move prometheus as an add-on i.e not mandated but advised.
+This includes moving the linkerd-prometheus manifests as a sub-chart and
+allowing more configuration (more on this later), along with documentation
+and tooling to use Linkerd with an existing prometheus installation.
+
+## Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+Lots of users already have a exisiting prometheus installation in their
+cluster and don't like managing another prometheus instance especially
+because they could be resource intensive and painful. They usually
+have a prometheus installation paired up with a long term monitroing
+solutions like Cortex,InfluxDB, or third party solutions like Grafana
+Cloud, etc. In those cases prometheus is just a temporary component, that
+keeps pushing metrics into their long term solution, making users want to
+re-use their exisitng prometheus.
+
+The following are the list of problems that we plan to solve:
+
+- Hard requirement of linkerd-prometheus.
```

Comment:
What's the actual problem that this solves, though?  Why is hard requirement of linkerd-prometheus bad?

Reply:
This is a very good decision technically as there are clear separation of concerns here but a lot of people consider it as a shortcoming and have been asking the bringing their own prometheus use-case.